pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -4,33 +4,77 @@ fixupOutputHooks+=(_doStrip)
 
 _doStrip() {
     if [ -z "$dontStrip" ]; then
-        stripDebugList=${stripDebugList:-lib lib32 lib64 libexec bin sbin}
-        if [ -n "$stripDebugList" ]; then
-            stripDirs "$stripDebugList" "${stripDebugFlags:--S}"
+        if [ -n "${stripDebugList:-*}" ]; then
+            stripDirs "${stripDebugList:-*}" "${stripDebugFlags:--S}"
         fi
 
-        stripAllList=${stripAllList:-}
-        if [ -n "$stripAllList" ]; then
-            stripDirs "$stripAllList" "${stripAllFlags:--s}"
+        if [ -n "${stripAllList:-}" ]; then
+            stripDirs "${stripAllList:-}" "${stripAllFlags:--s}"
         fi
     fi
 }
 
 stripDirs() {
-    local dirs="$1"
     local stripFlags="$2"
-    local dirsNew=
 
-    for d in ${dirs}; do
-        if [ -d "$prefix/$d" ]; then
-            dirsNew="${dirsNew} $prefix/$d "
+    # Ensure $prefix is a directory
+    if [ -f "$prefix" ]; then
+        header "Stripping (with flags $stripFlags) the file ${prefix}."
+        echo -ne $prefix'\0' | stripAll "$stripFlags"
+        stopNest
+    fi
+    [ -d "$prefix" ] || return 0
+
+    # Ensure expansion occurs within $prefix/.
+    pushd "$prefix" >/dev/null
+        GLOBIGNORE="" local stripDirs=( $1 )
+        GLOBIGNORE="" local ignoreDirs=( $dontStripList )
+    popd >/dev/null
+    local debugIncluded= debugExcluded= p=
+
+    # `find` fails on missing paths. Remove them.
+    local roots=()
+    for p in "${stripDirs[@]}"; do
+        if [ -e "$prefix/$p" ]; then
+            roots+=( "$prefix/$p" )
+            debugIncluded="$debugIncluded${debugIncluded:+,}$p"
         fi
     done
-    dirs=${dirsNew}
+    [ "${#roots[@]}" -eq 0 ] && return 0
```

Comment:
This line will fail if the length is nonzero... which clearly wasn't intended (IIRC we run in mode where failing line aborts everything). Let me suggest `[ "${#roots[@]}" -gt 0 ] || return 0`